src/client/reducers/reducer_movies.js

```diff
@@ -0,0 +1,29 @@
+import C from '../actions/constants';
+
+function findIndex(movie, state) {
+    return state.findIndex((item) => {
+        return movie.unit === item.unit;
+    });
+}
+
+export default function (state = [], action) {
+    switch (action.type) {
+        case C.POPULATE_MOVIES:
+            if (Array.isArray(action.payload)) {
+                const filtered = action.payload.filter((item) => {
+                    console.log(findIndex(item, state) < 0);
+                    return findIndex(item, state) < 0;
+                });
+                const merged = [...state, ...filtered];
+                const allIds = merged.map((item) => {
+                    return item.unit;
+                });
+                console.log({ byId: merged, allIds });
+                return { byId: merged, allIds };
+            } return [action.payload];
```

Comment:
@vitkuz normally here we return `state`.

Reply:
@vitkuz we have a store and it's our only source of truth. In general it is possible to return action.payload but it's some kind of exceptional case.

In this scenario when you add movies to the store you should return the store and on components side select necessary movies and that's it.

I know that in this specific case returning action.payload seems more natural but i would really appreciate if we check the common flow on at lease one example.

Reply:
@algbeta Fixed